packages/dbml-core/src/parse/ANTLR/ASTGeneration/PostgresASTGen.js

```diff
@@ -0,0 +1,941 @@
+      const secondFieldNames = ctx.opt_column_list().accept(this)?.map(c => c.value);
+
+      const actions = ctx.key_actions().accept(this);
+      return {
+        kind: 'inline_ref',
+        value: {
+          endpoints: [{
+            tableName: null,
+            schemaName: null,
+            fieldNames: null,
+            relation: '*',
+          }, {
+            tableName: refTableName,
+            schemaName: refSchemaName,
+            fieldNames: secondFieldNames,
+            relation: '1',
+          }],
+          onDelete: actions.onDelete,
+          onUpdate: actions.onUpdate,
+        }
+      }
+    }
+  }
+
+  // check PostgresSQLParser.g4 line 3619
+  visitB_expr (ctx) {
+    if (ctx.c_expr()) return ctx.c_expr().accept(this);
+    return {
+      value: ctx.getText(),
+      type: 'expression',
+    }
+  }
+
+  // check PostgresSQLParser.g4 line 3640
+  visitC_expr_expr (ctx) {
+    if (ctx.aexprconst()) return ctx.aexprconst().accept(this);
+    if (ctx.a_expr()) return {
+      value: ctx.a_expr().getText(),
+      type: 'expression'
+    }
+    return {
+      value: ctx.getText(),
+      type: 'expression',
+    }
+  }
+
+  visitC_expr_exists (ctx) {
+    return {
+      value: ctx.getText(),
+      type: 'expression',
+    }
+  }
+
+  visitC_expr_case (ctx) {
+    return {
+      value: ctx.getText(),
+      type: 'expression',
+    }
+  }
+
+  // iconst | fconst | sconst | bconst | xconst | func_name (sconst | OPEN_PAREN func_arg_list opt_sort_clause CLOSE_PAREN sconst) | consttypename sconst | constinterval (sconst opt_interval | OPEN_PAREN iconst CLOSE_PAREN sconst) | TRUE_P | FALSE_P | NULL_P
+  visitAexprconst (ctx) {
+    if (ctx.sconst() && ctx.getChildCount() === 1) return {
+      value: ctx.sconst().accept(this),
+      type: 'string',
+    }
+
+    if (ctx.TRUE_P() || ctx.FALSE_P() || ctx.NULL_P()) return {
+      value: ctx.getText(),
+      type: 'boolean',
+    }
+
+    if (ctx.iconst() || ctx.fconst) return {
+      value: ctx.getText(),
+      type: 'number',
+    }
+
+    return {
+      value: ctx.getText(),
+      type: 'expression',
+    }
+  }
+
+  // key_update | key_delete | key_update key_delete | key_delete key_update |
+  visitKey_actions (ctx) {
+    let [onDelete, onUpdate] = [null, null];
+    if (ctx.key_update()) onUpdate = ctx.key_update().accept(this);
+    if (ctx.key_delete()) onDelete = ctx.key_delete().accept(this);
+    return {
+      onDelete,
+      onUpdate,
+    };
+  }
+
+  // ON UPDATE key_action
+  visitKey_update (ctx) {
+    return ctx.key_action().accept(this);
+  }
+
+  // ON DELETE_P key_action
+  visitKey_delete (ctx) {
+    return ctx.key_action().accept(this);
+  }
+
+  // NO ACTION | RESTRICT | CASCADE | SET (NULL_P | DEFAULT)
+  visitKey_action (ctx) {
+    let r = ctx.getChild(0).getText();
+    for (let i =  1; i < ctx.getChildCount(); i++) {
+      r += ` ${ctx.getChild(i).getText()}`;
+    };
```

Comment:
Can we use lodash functions here?

Reply:
Updated in https://github.com/holistics/dbml/pull/416/commits/15d016f623f2d2c5bcf6a5ab90510d1f4723b782. Although not using `lodash`, I replace the for loop and made it functional by generating an array range of indices and then reduce it into a string.